From e914ec9c8e498b1a568bd5eaa71028a2afe3d8f7 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 2 Oct 2018 00:09:57 +0530
Subject: [PATCH 0435/1852] drm: xlnx: mixer: Fix switching to 4K after 1080p

commit 47fe47a92ad1e1fd9e025a5e83964c4a45a7fae1 from
https://github.com/Xilinx/linux-xlnx.git

This patch fixes the issue of switching to higher resolution like 4K
once the primary layer has been set to a lower resolution like 1080p.
This is done by disabling the need to check frame parameters like width,
height if the plane to be set is the primary plane.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Saurabh Sengar <saurabhs@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_mixer.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_mixer.c b/drivers/gpu/drm/xlnx/xlnx_mixer.c
index a41d2951504b..aaf62f6b074c 100644
--- a/drivers/gpu/drm/xlnx/xlnx_mixer.c
+++ b/drivers/gpu/drm/xlnx/xlnx_mixer.c
@@ -1747,6 +1747,12 @@ static int xlnx_mix_plane_atomic_check(struct drm_plane *plane,
 	int scale;
 	struct xlnx_mix_plane *mix_plane = to_xlnx_plane(plane);
 	struct xlnx_mix_hw *mixer_hw = to_mixer_hw(mix_plane);
+	struct xlnx_mix *mix;
+
+	/* No check required for the drm_primary_plane */
+	mix = container_of(mixer_hw, struct xlnx_mix, mixer_hw);
+	if (mix->drm_primary_layer == mix_plane)
+		return 0;
 
 	scale = xlnx_mix_get_layer_scaling(mixer_hw,
 					   mix_plane->mixer_layer->id);
-- 
2.31.1

